src/core/quantum_circuit.jl

```diff
@@ -24,12 +24,10 @@ Base.@kwdef struct QuantumCircuit
     function QuantumCircuit(qubit_count::Int,gates::Vector{GateType}) where GateType <: Gate
         @assert qubit_count>0 ("$(:QuantumCircuit) constructor requires qubit_count>0. Received: $qubit_count")
     
-        c=new(qubit_count,[])
-
-        # add gates, with ensure_gate_is_in_circuit()
-        append!(c,gates)
```

Comment:
I have no objection to the change, but I dont see why this no longer works

Reply:
It failed precompilation using Julia 1.9 when calling `Pkg.test()`. It might have been a bug in the precompile tools, but since the fix was so simple, I didn't spend time investigating further.